### PR TITLE
build docs on each PR

### PR DESCRIPTION
**Pull request by techknowlogick**
It is minimal extra CI time to build docs on each PR.

### CODE REVIEW

**Comment by silverwind** [comment]
Related: https://github.com/go-gitea/gitea/pull/24877

File: .github/workflows/pull-compliance-docs.yml
Comment (yardenshoham) [inline .github/workflows/pull-compliance-docs.yml]:
```suggestion
on: ["pull_request"]
```

**Comment by techknowlogick** [comment]
> Likely conflicted by #24877.

Yeah, this is a stopgap until that could be merged.

**Comment by silverwind** [comment]
I think https://github.com/go-gitea/gitea/pull/24877 is ready for a merge and a few test runs, but happy to merge this as well if it takes longer.

**Review by silverwind — APPROVED** [review]

**Pull request closed without merging**
